### PR TITLE
Fix wrong file revision chosen in file diff

### PR DESCRIPTION
**Pull request by eseliger**
The merge base is the commit that the diff applies to, not the specified base OID. Hence, we need to find the merge-base and use that as the base in the file.

### CODE REVIEW

File: internal/vcs/git/exec.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ var (
 		"rev-list":     {"--max-parents", "--reverse", "--max-count"},
 		"ls-remote":    {"--get-url"},
 		"symbolic-ref": {"--short"},
+		"merge-base":   {},
```
Comment (eseliger) [inline internal/vcs/git/exec.go:121]:
```suggestion
```

File: cmd/frontend/graphqlbackend/repository_comparison.go [hunk 2]
```diff
@@ -106,7 +106,17 @@ func NewRepositoryComparison(ctx context.Context, r *RepositoryResolver, args *R
 	wg.Add(2)
 	go func() {
 		defer wg.Done()
-		base, baseErr = getCommit(ctx, *grepo, baseRevspec)
+		// Find the common merge-base for the diff. That's what the diff will actually need to be applied to,
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/repository_comparison.go:109]:
```suggestion
		// Find the common merge-base for the diff. That's what the diff was applied to,
```

File: cmd/frontend/graphqlbackend/repository_comparison.go [hunk 3]
```diff
@@ -106,7 +106,17 @@ func NewRepositoryComparison(ctx context.Context, r *RepositoryResolver, args *R
...
+		// not the baseRevspec.
+		mergeBaseCommit, err := git.MergeBase(ctx, *grepo, api.CommitID(baseRevspec), api.CommitID(headRevspec))
+		if err != nil {
+			baseErr = err
+			return
+		}
+		// We use the merge-base as the base commit here, as the diff will only be guaranteed to be
```
Comment (mrnugget) [inline cmd/frontend/graphqlbackend/repository_comparison.go:116]: Does that comment also make sense when not seen in the context of campaigns?

Comment (eseliger) [reply cmd/frontend/graphqlbackend/repository_comparison.go:116]: I think so, for all repo-comparisons we generate a diff.

**Pull request merged**